lib/widgets/action_sheet.dart

```diff
@@ -41,21 +44,48 @@ void showMessageActionSheet({required BuildContext context, required Message mes
       && reactionWithVotes.userIds.contains(store.selfUserId))
     ?? false;
 
-  showDraggableScrollableModalBottomSheet<void>(
+  final optionButtons = [
+    if (!hasThumbsUpReactionVote)
+      AddThumbsUpButton(message: message, messageListContext: context),
+    StarButton(message: message, messageListContext: context),
+    if (isComposeBoxOffered)
+      QuoteAndReplyButton(message: message, messageListContext: context),
+    if (showMarkAsUnreadButton)
+      MarkAsUnreadButton(message: message, messageListContext: context, narrow: narrow),
+    CopyMessageTextButton(message: message, messageListContext: context),
+    CopyMessageLinkButton(message: message, messageListContext: context),
+    ShareButton(message: message, messageListContext: context),
+  ];
+
+  showModalBottomSheet<void>(
     context: context,
+    // Clip.hardEdge looks bad; Clip.antiAliasWithSaveLayer looks pixel-perfect
+    // on my iPhone 13 Pro but is marked as "much slower":
+    //   https://api.flutter.dev/flutter/dart-ui/Clip.html
+    clipBehavior: Clip.antiAlias,
+    useSafeArea: true,
+    isScrollControlled: true,
     builder: (BuildContext _) {
-      return Column(children: [
-        if (!hasThumbsUpReactionVote)
-          AddThumbsUpButton(message: message, messageListContext: context),
-        StarButton(message: message, messageListContext: context),
-        if (isComposeBoxOffered)
-          QuoteAndReplyButton(message: message, messageListContext: context),
-        if (showMarkAsUnreadButton)
-          MarkAsUnreadButton(message: message, messageListContext: context, narrow: narrow),
-        CopyMessageTextButton(message: message, messageListContext: context),
-        CopyMessageLinkButton(message: message, messageListContext: context),
-        ShareButton(message: message, messageListContext: context),
-      ]);
+      return SafeArea(
+        minimum: const EdgeInsets.only(bottom: 16),
+        child: Padding(
+          padding: const EdgeInsets.fromLTRB(16, 0, 16, 0),
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.stretch,
+            mainAxisSize: MainAxisSize.min,
+            children: [
+              // TODO(#217): show message text
+              Flexible(child: InsetShadowBox(
+                top: 8, bottom: 8,
+                color: DesignVariables.of(context).bgContextMenu,
+                child: SingleChildScrollView(
+                  padding: const EdgeInsets.only(top: 16, bottom: 8),
+                  child: ClipRRect(
+                    borderRadius: BorderRadius.circular(7),
+                    child: Column(spacing: 1, children: optionButtons),
+                  )))),
```

Comment:
nit:
```suggestion
                    child: Column(spacing: 1,
                      children: optionButtons))))),
```

* our standard close-paren formatting
* and perhaps more importantly: keeps `children: optionButtons`, which is the ultimate payload that contains the meaningful content which everything else here is wrapping, visible on its own line